fixing navbar  + fetch data in homepage from DB

razashtar93 opened the pull request:
- now navbar present log in and register when user isn't logged in
- after user log in the navbar present log out option
- also added  button contact us to display in the navbar
- changed contact_us.html to extend app_layout 
- changed logo in README.md - this is the final logo that need to be there

Yarboa commented on templates/landing/homepage.html:
```diff
@@ -17,7 +17,7 @@ <h1 class="font-weight-bold">Welcome To</h1>
           <h3 class="my-2">Choose University</h3>
           <div>
-            <select aria-label="choose-university" id="selectContent">
+            <select aria-label="choose-university" id="select_Content_university">
```
It is the 5 time i am asking, 
Can you read universities from from DB? 
and through a view parse the data to template :) ?

razashtar93 replied:
> It is the 5 time i am asking, Can you read universities from from DB? and through a view parse the data to template :) ?

well this is part of the problems we are having ..  :sweat_smile: 
once we understand how to do it correctly and make it work we will send a PR .. @Yarboa

DanielItkes replied:
@Yarboa We can make the page read the data from the database but we don't know how to make it send us to the desird page..
For example -
Choosing a university from a drop down menu -> automaticaly get redirected to university page (aka 'university_name.html'

Yarboa replied:
Please refer this
https://spookylukey.github.io/django-views-the-right-way/context-data.html
Try to think how you query db calling the template and run for loop on date returned

Yarboa replied:
@razashtar93 any new commits on the road? 
Please add also tests with client fixture for the contact_us page

razashtar93 replied:
@Yarboa please see what I wrote on slack

Yarboa replied:
> * now navbar present log in and register when user isn't logged in

@razashtar93 So please add it to the Description

razashtar93 commented:
## Now I made homepage fetch data from DB
homepage.html get universities and degrees from what we have in the DB

Update homepage url !

Created function in AcademeData/views that fetch universities and degrees
from DB

Also changed something small in contact_us

Yarboa commented on AcadeMeData/views.py:
```diff
@@ -16,3 +17,17 @@ def register(request):
+def universities_degrees(request):
+    all_universities = University.objects.all()
+    all_degrees = Degree.objects.all()
+    #     University.get_universities() # added this function in models.py
```
Please remove

Yarboa commented on AcadeMeData/views.py:
```diff
@@ -16,3 +17,17 @@ def register(request):
+    #     University.get_universities() # added this function in models.py
+    context = {'all_universities': all_universities, 'all_degrees': all_degrees}
+    return render(request, '../templates/landing/homepage.html', context)
+
+# def degrees(request):
+#     all_degrees = Degree.objects.all()
+#
+#     context = {'all_degrees': all_degrees}
+#     return render(request, '../templates/landing/homepage.html', context)
```
Please remove

Yarboa reviewed (commented):
Please fix the tests as per this
https://pastebin.com/WnHdXL8t

You can assert the page content also

I think that in migration file you should fix this also
https://pastebin.com/PzbLkKGv

Omeramsc commented on AcadeMe/urls.py:
```diff
@@ -18,12 +18,12 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path('', views.homePage,  name='homepage'),
+    # path('', views.homePage, name='homepage'),
```
if the comment serves no purpose, remove it.

Omeramsc reviewed (changes requested):
I see a lot of changes that are not relevant to the navbar... why is it here?

razashtar93 commented:
> I see a lot of changes that are not relevant to the navbar... why is it here?

Started by fixing something in the navbar, then Yariv asked to change the homepage that it will fetch universities and degrees from the DB, and now I've got a lot of errors from the tests (and also I think there's a problem with some of the migration files)

Yarboa commented:
> * now navbar present log in and register when user isn't logged in

Can you suggest a test here ^^
> * after user log in the navbar present log out option

Can you suggest a test here ^^
> * also added  button contact us to display in the navbar

Can you suggest a test here ^^
> * changed contact_us.html to extend app_layout
> * changed logo in README.md - this is the final logo that need to be there

Don you think to add here in the description the:
Dynamic fetch universities coursed in home page?

razashtar93 commented:
> > * now navbar present log in and register when user isn't logged in
> 
> Can you suggest a test here ^^
> 
> > * after user log in the navbar present log out option
> 
> Can you suggest a test here ^^
> 
> > * also added  button contact us to display in the navbar
> 
> Can you suggest a test here ^^
> 
> > * changed contact_us.html to extend app_layout
> > * changed logo in README.md - this is the final logo that need to be there
> 
> Don you think to add here in the description the: Dynamic fetch universities coursed in home page?


OK, so I added relevant tests, few problems:
1. I made navbar links in app_layout more generic - but nothing is displayed in the navbar
2. for some reason pytest is not recognize my tests (I added another existing test to the file and it run so I don't really understand what the problem there) 
3. test for contact us will come with Dan PR
4. the fetch of universities and degrees in homepage in not ready yet. it just displayed all the universities and degrees in the DB
5. Good Night.

Yarboa commented on templates/app_layout.html:
```diff
@@ -1,47 +1,67 @@
+            {% endfor %}
+
+
+            <!--            {% if not user.is_authenticated %}-->
```
Do you plan to use it? 
In case not, can you please delete it ?

Yarboa commented on AcadeMe/views.py:
```diff
@@ -19,3 +28,12 @@ def loginPage(request):
     return render(request, 'registration/registration.html')
+
+
+# def add_navbar_links_to_context(request):
```
Why is it here? 
Do you plane to use it? in case not, please remove this part

Yarboa commented on AcadeMeData/tests/test_navbar_links.py:
```diff
@@ -0,0 +1,27 @@
+
+
+@pytest.fixture
+def user_example():
+    user_data = {'username': "username", 'password': "password", 'email': "user@example.com",
+                 'university': "RU", 'degree': "CS"}
+    user = User.create_user(*user_data)
+    return user
+
```
maybe this should move to? https://github.com/redhat-beyond/AcadeMe/blob/main/AcadeMeData/conftest.py

or at least open an Issue related to moving user_example over there? and than suggest a new PR?

Omeramsc reviewed (approved):
LGTM, but won't merge until Yariv's comments will get resolved

razashtar93 commented:
> LGTM, but won't merge until Yariv's comments will get resolved

All problems solved

Yarboa commented on AcadeMe/views.py:
```diff
@@ -1,16 +1,28 @@
 
 
 def app_layout(request):
+    # context = {}
+    # if request.user.is_authenticated:
+    #     context['navbar_links'] = {f"Welcome {request.user.username}": "#", "Logout": "/logout"}
+    # else:
+    #     context['navbar_links'] = {"Login": "/login", "Register": "/register"}
```
I think you do not want to Approve this PR :)
Please delete it

razashtar93 replied:
uhhh I forgot that .... :expressionless:

Done.
